Add lustre support to MPIFileUtils

bdevcich opened the pull request:
`dcp` has a `--grouplock` option and that requires the lustre client. Lustre is now being built in the builder stage of the image and is used to enable lustre build support in MPIFileUtils and also runtime support.

This image's OS is debian bookworm, which has deprecated `dpatch` - a required build dependency for older versions of lustre. The latest cray release of lustre 2.15 is being used, where `dpatch` is no longer required.

bdevcich committed:
Add lustre support to MPIFileUtils

`dcp` has a `--grouplock` option and that requires the lustre client.
Lustre is now being built in the builder stage of the image and is used
to enable lustre build support in MPIFileUtils and also runtime support.

This image's OS is debian bookworm, which has deprecated `dpatch` - a
required build dependency for older versions of lustre. The latest cray
release of lustre 2.15 is being used, where `dpatch` is no longer
required.

Signed-off-by: Blake Devcich <blake.devcich@hpe.com>

roehrich-hpe commented on Makefile:
```diff
@@ -18,24 +18,40 @@
+
+# Only support linux/amd64 builds. This reduces complexity with installing
+# lustre, which is an option enabled for mpifileutils.
+PLATFORM ?= linux/amd64
```
Does this mean we can't build a Mac version for use in Kind anymore?

bdevcich replied:
I believe I can drop this explicit `PLATFORM` definition and with the recent additions that I've made, we can build for arm as well. I'll test this out more.

roehrich-hpe commented on Makefile:
```diff
@@ -18,24 +18,40 @@
+CONTAINER_TOOL ?= docker
+
+# Which tag to pull from the cray lustre source code.
+# **Note:** This must also be set in .github/workflows/main.yaml for both
```
Set it the same way I set OPENMPI_VERSION so you don't have to set it in the workflow.

bdevcich replied:
Thanks.

bdevcich committed:
Review comments:
- Add default value for LUSTRE_VERSION
- Don't set lustre version in Makefile/workflow
- Remove explicit setting of platform so that arm64 is also supported
